### ISSUE TITLE
provide Combinators for Datum

### ISSUE THREAD

**Issue by tippenein** [mbeidler/ghcjs-c3#2]
The existing `Column` type for input is _ok_ but we need more sophisticated combinators for data input.

Context: https://github.com/mbeidler/ghcjs-c3/pull/1#issuecomment-287929083

My initial impression was providing lens' / prisms around a couple different data representations and ultimately converting them to "columns" in the json.

There are also the `json` and `url` data input types (http://c3js.org/reference.html#data-url)

**Comment by mbeidler** [comment]
I agree. There are a few different APIs we could consider. As you mentioned it would also be good to have high-level functions for simple use-cases. For example, to generate a simple scatter plot provided lenses from s to x and s to y, we could have a signature like:

```Haskell
xy :: Getter s Double -> Getter s Double -> [s] -> IO Chart
```

It would be also worth thinking about what low-level API to use internally for optimal code reuse, since there are a large number of optional configuration parameters and ideally we don't want to clutter up our types with `Maybe a` values everywhere.
